src/metabase/driver/postgres.clj

```diff
@@ -880,6 +883,137 @@
           "from table_privileges t"]))
        (filter #(or (:select %) (:update %) (:delete %) (:update %)))))
 
+(def ^:private table-type-clauses
+  {"TABLE"              [:and
```

Comment:
As someone who has also been doing a bit of SQL -> HoneySQL translation myself lately, I've come to agree. I imagined there might be some point in the future where we want to do more sophisticated string manipulation, and HoneySQL can help. But translating SQL to HoneySQL is mostly a waste of time in this type of scenario. Since this has been done already though there's no need to undo it. The `:inline` repetition doesn't bother me too much on these lines since it's all neatly aligned.
https://github.com/metabase/metabase/pull/39670/files#diff-122d8f1486c5b97fd092fb98e7419b767c12d378bc6e668c16b1041cd965f9cbR971-R1000

I do think `table-type-clauses` is unnecessary though, and can be deduped with the logic on lines 971-1000. 

The only types we care about are these: `"TABLE" "PARTITIONED TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW" "EXTERNAL TABLE" "DYNAMIC_TABLE"`

I think we just need to excluding system tables with 
`[:<> :n.nspname [:inline "information_schema"]]` and `[(keyword "!~") :n.nspname [:inline "^pg_"]]`.

and then use CTEs to filter on `:type`, generated from this part of the case expression https://github.com/metabase/metabase/pull/39670/files#diff-122d8f1486c5b97fd092fb98e7419b767c12d378bc6e668c16b1041cd965f9cbR991-R1001

Reply:
yea I guess I was too focused on converting them. simplified and use raw SQL for types.